File: lib/salus-emul.php
<?php


define("EMUL_DEVICES_ONLINE_FILE",'/local/fakes/devices_online.html');
define("EMUL_DEVICES_OFFLINE_FILE",'/local/fakes/devices_offline.html');


// args:
// return: 
//   true=success or false=failed
function emul_load_devices()
{
	global $app;
	
	$file_name = $app->home_path().($app->salus->emul_submode()==SalusConnect::EMUL_ONLINE? EMUL_DEVICES_ONLINE_FILE: EMUL_DEVICES_OFFLINE_FILE);
	log_debug('emul_load_devices. file name='.$file_name);

	if (!file_exists($file_name)) {
		log_error('No '.$file_name.' file');
		return false;
	}

	$content = file_get_contents($file_name);
	if ($content === false) {
		log_error('Can not load '.$file_name);
		return false;
	}

	log_ok('Loaded '.$file_name);
	return $content;

}


?>

File: lib/salus-emul.py
import os.path

from utils.py import *

EMUL_DEVICES_ONLINE_FILE = "/local/fakes/devices_online.html"
EMUL_DEVICES_OFFLINE_FILE = "/local/fakes/devices_offline.html"


## args:
## return: 
##   true=success or false=failed
def emul_load_devices():

	file_path = app.home_path()+(app.salus.emul_submode()==SalusConnect.EMUL_ONLINE? EMUL_DEVICES_ONLINE_FILE: EMUL_DEVICES_OFFLINE_FILE)
	log_debug('emul_load_devices. file path='+file_path);

	if not os.path.isfile(file_path):
		log_error('No '+file_path+' file')
		return False

	try:
		with open (file_path, 'r') as fp:
			content = fp.read

	log_ok('Loaded '+file_path)
	
	return content
